src/grt/src/fastroute/src/maze3D.cpp

```diff
@@ -1079,7 +1081,8 @@ void FastRouteCore::mazeRouteMSMDOrder3D(int expand, int ripupTHlb, int ripupTHu
                 tmp = d1_3D_[curL][curY][curX] + 1;
                 tmpX = curX + 1;  // the right neighbor
 
-                if (h_edges_3D_[grid].usage < h_edges_3D_[grid].cap) {
+                if (h_edges_3D_[grid].usage < h_edges_3D_[grid].cap &&
+                    curL >= net->minLayer && curL <= net->maxLayer) {
```

Comment:
It's quite minor but I prefer

net->minLayer <= curL && curL <= net->maxLayer

as it reads more like the mathematical

net->minLayer <= curL  <= net->maxLayer

You can change it or not as you prefer.

Reply:
Makes sense, I'll change it and keep it in mind for future updates.